package-dev/Plugins/macOS/SentryNativeBridge.m

```diff
@@ -0,0 +1,246 @@
+{
+    NSMutableDictionary *dictOptions = (__bridge NSMutableDictionary *)options;
+    dictOptions[[NSString stringWithUTF8String:name]] = [NSNumber numberWithInt:value];
+}
+
+void SentryNativeBridgeStartWithOptions(const void *options)
+{
+    NSMutableDictionary *dictOptions = (__bridge_transfer NSMutableDictionary *)options;
+    [SentrySDK performSelector:@selector(startWithOptions:) withObject:dictOptions];
+}
+
+/*******************************************************************************/
+/* The remaining code is a copy of iOS/SentryNativeBridge.m with changes to    */
+/* make it work with dynamically loaded classes. Mainly:                       */
+/*   - call: [class performSelector:@selector(arg1:arg2:)                      */
+/*                  withObject:arg1Value withObject:arg2Value];                */
+/*   - use `id` as variable types                                              */
+/*   - use [obj setValue:value forKey:@"prop"] instead of `obj.prop = value`   */
+/*******************************************************************************/
+
+int SentryNativeBridgeCrashedLastRun()
+{
+    @try {
+        return [SentrySDK performSelector:@selector(crashedLastRun)] ? 1 : 0;
+    } @catch (NSException *exception) {
+        NSLog(@"Sentry (bridge): failed to get crashedLastRun: %@", exception.reason);
+    }
+    return -1;
+}
+
+void SentryNativeBridgeClose()
+{
+    @try {
+        [SentrySDK close];
+    } @catch (NSException *exception) {
+        NSLog(@"Sentry (bridge): failed to close: %@", exception.reason);
+    }
+}
+
+void SentryNativeBridgeAddBreadcrumb(
+    const char *timestamp, const char *message, const char *type, const char *category, int level)
+{
+    if (timestamp == NULL && message == NULL && type == NULL && category == NULL) {
+        return;
+    }
+
+    // declaring the (block) callback as a variable to avoid too much editor blank space on the left
+    void (^scopeUpdateBlock)(id) = ^void(id scope) {
+        id breadcrumb = [[SentryBreadcrumb alloc] init];
+
+        if (timestamp != NULL) {
+            NSDateFormatter *dateFormatter = [[NSDateFormatter alloc] init];
+            [dateFormatter setDateFormat:NSCalendarIdentifierISO8601];
+            [breadcrumb
+                setValue:[dateFormatter dateFromString:[NSString stringWithUTF8String:timestamp]]
+                  forKey:@"timestamp"];
+        }
+
+        if (message != NULL) {
+            [breadcrumb setValue:[NSString stringWithUTF8String:message] forKey:@"message"];
+        }
+
+        if (type != NULL) {
+            [breadcrumb setValue:[NSString stringWithUTF8String:type] forKey:@"type"];
+        }
+
+        if (category != NULL) {
+            [breadcrumb setValue:[NSString stringWithUTF8String:category] forKey:@"category"];
+        }
+
+        [breadcrumb setValue:[NSNumber numberWithInt:level] forKey:@"level"];
+
+        [scope performSelector:@selector(addBreadcrumb:) withObject:breadcrumb];
```

Comment:
So no way we can add the header and use the actual types? I assume not since you went with this approach.

Reply:
I couldn't get it to work. That being said, maybe there's a way, I just don't know it :)